React/Base/RCTImageSource.m

```diff
@@ -57,13 +59,31 @@ + (RCTImageSource *)RCTImageSource:(id)json
   }
 
   NSURL *imageURL;
+  NSDictionary *headers = nil;
   CGSize size = CGSizeZero;
   CGFloat scale = 1.0;
   BOOL packagerAsset = NO;
   if ([json isKindOfClass:[NSDictionary class]]) {
     if (!(imageURL = [self NSURL:RCTNilIfNull(json[@"uri"])])) {
       return nil;
     }
+    headers = [self NSDictionary:RCTNilIfNull(json[@"headers"])];
+#if RCT_DEBUG
```

Comment:
Think we want to remove this so that we nil out `headers` even in production when receiving invalid values. The validation check is cheap and bad errors occurred at FB when the iOS app shipped with code that was stricter during dev than production because the code paths were different. Better to crash or not crash the same way everywhere.